lib/portage/dbapi/bintree.py

```diff
@@ -2199,6 +2173,41 @@ def _allocate_filename_multi(self, cpv):
             else:
                 return filename
 
+    def _allocate_build_id(self, cpv):
+        # First, get the max build_id found when _populate was
+        # called.
+        max_build_id = self._max_build_id(cpv)
+
+        # A new package may have been added concurrently since the
+        # last _populate call, so use increment build_id until
+        # we locate an unused id.
+        pf = catsplit(cpv)[1]
+        build_id = max_build_id + 1
+
+        try:
+            binpkg_format = cpv.binpkg_format
+        except AttributeError:
+            binpkg_format = self.settings.get(
+                "BINPKG_FORMAT", SUPPORTED_GENTOO_BINPKG_FORMATS[0]
+            )
+
+        if binpkg_format == "xpak":
+            filename_format = "%s-%s.xpak"
+        elif binpkg_format == "gpkg":
+            filename_format = "%s-%s.gpkg.tar"
+        else:
+            raise InvalidBinaryPackageFormat(binpkg_format)
+
+        while True:
+            filename = filename_format % (
```

Comment:
These days we really prefer `.format()` for new code.

Reply:
Was reused old code. Do you prefer new `f"{var}"` or `.format()`.

Reply:
`f`, I suppose. Make the `if` above set `suffix` rather than the full format string, and use that.